src/Debug/skeletonViewer.ts

```diff
@@ -324,8 +585,18 @@ export class SkeletonViewer {
                 scene.animationsEnabled = false;
             }
 
-            if (this.options.returnToRest) {
-                this.skeleton.returnToRest();
+            if (this.options.currentStateIsRestPose) {
+                let _s = Vector3.Zero();
+                let _r = Quaternion.Identity();
+                let _t = Vector3.Zero();
+                this.skeleton.bones.forEach((b) => {
+                    b.getLocalMatrix().decompose(_s, _r, _t);
+                    b.setRestPose(Matrix.Compose(_s, _r, _t));

```

Comment:
Why do we need this option in the first place?

It's something that the caller can do themselves if they want to change the rest pose, I feel it's not the skeleton viewer responsibility to do that. The skeleton viewer simply only you to view the skeleton / bones, not to modify it / its internal state.